### PR TITLE
FTUE: Add and display help center videos

### PR DESCRIPTION
**Pull request by embarks**
## Context
- Follow up PR to the [animated skeleton](https://github.com/google/web-stories-wp/pull/6158) which includes the MP4s provided by design and displays them in the quick tip menu.
- Need this display in order to finish [quick tip screens](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/5886)
- Figma reference [under FTUE](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2%3A91941)

## Summary
- add paths to relevant images to tip constants
- `useConfig` to get asset path
- add appropriate controls to video (e.g. use html5 video, muted needed for autoplay)

https://user-images.githubusercontent.com/41136059/106676058-7431d780-6573-11eb-81b7-f146fe75887d.mp4

## Relevant Technical Choices
- mp4s were chosen over lottie files to avoid installing a new library needed to display them, and the videos are small, saving on load time and project size

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Turn on Quick Tips/First Time User Experience experiements
2. Open Quick Tips in Editor
3. See beautiful animations play automatically

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

### Does this PR have a security-related impact?
- no

### Does this PR change what data or activity we track or use?
- no

### Does this PR have a legal-related impact?
- no

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #5586

### CODE REVIEW

**Comment by swissspidy** [comment]
If we expect the number of videos to grow over time, we should probably move them to our CDN instead of bundling them with the plugin. Otherwise that's gonna increase bundle size a lot.

**Comment by swissspidy** [comment]
> mp4s were chosen over lottie files to avoid installing a new library needed to display them, and the videos are small, saving on load time and project size

I think we also discussed using webm files because of transparency? Not needed anymore?

**Comment by BrittanyIRL** [comment]
> If we expect the number of videos to grow over time, we should probably move them to our CDN instead of bundling them with the plugin. Otherwise that's gonna increase bundle size a lot.

We just have these 7 for now, would it be worthwhile to add a separate ticket to pull these to the CDN?

**Comment by BrittanyIRL** [comment]
> > mp4s were chosen over lottie files to avoid installing a new library needed to display them, and the videos are small, saving on load time and project size
> 
> I think we also discussed using webm files because of transparency? Not needed anymore?

Asking the Huge team to verify.

**Comment by embarks** [comment]
> > > mp4s were chosen over lottie files to avoid installing a new library needed to display them, and the videos are small, saving on load time and project size
> > 
> > I think we also discussed using webm files because of transparency? Not needed anymore?
> 
> Asking the Huge team to verify.

Forgot about this, I'll use the webm files

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 1]
```diff
@@ -44,7 +45,7 @@ const Overflow = styled.div`
 `;
 
 // @TODO update with actual figure.
```
Comment (samwhale) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: do we need this anymore?

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 2]
```diff
@@ -44,7 +45,7 @@ const Overflow = styled.div`
...
-const Figure = styled.div`
+const Figure = styled.video`
```
Comment (samwhale) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: TIL about the video element. This is dope

**Review by samwhale — APPROVED** [review]
Super pretty 😍. Will quality of the videos be a factor we should worry about? I guess it depends on the user's wifi connection

**Comment by swissspidy** [comment]
> > If we expect the number of videos to grow over time, we should probably move them to our CDN instead of bundling them with the plugin. Otherwise that's gonna increase bundle size a lot.
> 
> We just have these 7 for now, would it be worthwhile to add a separate ticket to pull these to the CDN?

I could live with that, as long as it's done before release.

**Comment by swissspidy** [comment]
> > If we expect the number of videos to grow over time, we should probably move them to our CDN instead of bundling them with the plugin. Otherwise that's gonna increase bundle size a lot.
> 
> We just have these 7 for now, would it be worthwhile to add a separate ticket to pull these to the CDN?

I could live with that, as long as it's done before release.

File: assets/src/edit-story/components/helpCenter/quickTip/index.js (hunk 2 above)
Comment (embarks) [reply assets/src/edit-story/components/helpCenter/quickTip/index.js]: [HTML5 is AWESOME](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5)

**Review by maxyinger — APPROVED** [review]
🎉

**Comment by BrittanyIRL** [comment]
> > > If we expect the number of videos to grow over time, we should probably move them to our CDN instead of bundling them with the plugin. Otherwise that's gonna increase bundle size a lot.
> > 
> > 
> > We just have these 7 for now, would it be worthwhile to add a separate ticket to pull these to the CDN?
> 
> I could live with that, as long as it's done before release.

@swissspidy Ticket! https://github.com/google/web-stories-wp/issues/6177

**Review by BrittanyIRL — APPROVED** [review]

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 3]
```diff
@@ -62,16 +62,22 @@ export function QuickTip({
...
     <Transitioner
       {...transitionProps}
       isLeftToRightTransition={isLeftToRightTransition}
     >
       <Panel>
         <Overflow>
-          <Figure />
+          {Boolean(figure) && (
+            <Figure controls={false} autoPlay loop muted>
+              <source src={`${assetsURL}/${figure}`} type="video/mp4" />
```
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: These are  webm files, not mp4s

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 4]
```diff
@@ -62,16 +62,22 @@ export function QuickTip({
   title,
   description,
   isLeftToRightTransition = true,
+  figure,
   ...transitionProps
 }) {
+  const { assetsURL } = useConfig();
```
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: You can use `cdnURL` now

File: assets/src/edit-story/components/helpCenter/constants.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ import { __ } from '@wordpress/i18n';
 export const TIPS = {
   addBackgroundMedia: {
     title: __('Add background media', 'web-stories'),
+    figure: 'images/editor/helpCenter/Add_BG_Module_1.webm',
```
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/constants.js]: For the CDN images, correct path will be `images/help-center/add_bg_module_1.webm`.

All lowercase.

Same for the other paths.

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 6]
```diff
@@ -62,16 +62,22 @@ export function QuickTip({
...
   return (
     <Transitioner
       {...transitionProps}
       isLeftToRightTransition={isLeftToRightTransition}
     >
       <Panel>
         <Overflow>
-          <Figure />
+          {Boolean(figure) && (
+            <Figure controls={false} autoPlay loop muted>
```
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: Probably want to add `noControls` prop as well. Doesn't make sense to show controls for this kind  of animation.

```suggestion
            <Figure controls={false} autoPlay loop muted noControls>
```

File: assets/src/edit-story/components/helpCenter/quickTip/index.js [hunk 7]
```diff
@@ -43,8 +44,7 @@ const Overflow = styled.div`
   padding: ${GUTTER_WIDTH}px;
 `;
 
-// @TODO update with actual figure.
-const Figure = styled.div`
+const Figure = styled.video`
```
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/quickTip/index.js]: Maybe rename to `Video`? Less confusing

**Comment by embarks** [comment]
The webms are not showing up as transparent right now. :) I pinged ws-design about it so just waiting on discussion around that. 👍

**Comment by swissspidy** [comment]
> The webms are not showing up as transparent right now. :) I pinged ws-design about it so just waiting on discussion around that. 👍

Hmm maybe they need to be re-exported again from Lottie to preserve the transparency, or converted using ffmpeg

**Comment by BrittanyIRL** [comment]
> The webms are not showing up as transparent right now. :) I pinged ws-design about it so just waiting on discussion around that. 👍

I think just for the sake of limiting spicy merge conflicts and unblocking work let's proceed with the existing webm files and replace them with the versions that have transparency when the design team gets them to us. Thoughts?

**Comment by maxyinger** [comment]
@BrittanyIRL I think that sounds like a good move. The current ticket makes it work for now, then we can make it better in successive ones.

We could also explore using srcsets and having design export multiple sizes. The figures should stay the same size in the browser, but it would allow for browsers to have some lower res options for slower networks. Also allow for browsers that don't support webm to have a fallback, which I think is more of a nice to have than a necessity with our current targeted browsers

**Comment by BrittanyIRL** [comment]
> @BrittanyIRL I think that sounds like a good move. The current ticket makes it work for now, then we can make it better in successive ones.
> 
> We could also explore using srcsets and having design export multiple sizes. The figures should stay the same size in the browser, but it would allow for browsers to have some lower res options for slower networks. Also allow for browsers that don't support webm to have a fallback, which I think is more of a nice to have than a necessity with our current targeted browsers

I'll make a follow up v2 ticket to explore srcsets for better perf! great idea. 

edit: ticket - https://github.com/google/web-stories-wp/issues/6190

**Comment by embarks** [comment]
@googlebot I consent.

**Comment by swissspidy** [comment]
@embarks as you're the PR author, you never have to worry about the bot :-) it's about who else has contributed to your PR — Max. He needs to reply to the bot.

**Comment by embarks** [comment]
> @embarks as you're the PR author, you never have to worry about the bot :-) it's about who else has contributed to your PR — Max. He needs to reply to the bot.

Ahhh, good to know. I got confused because once my git own author details got messed up and I had to amend the commits so the consent bot could see that I consented. This time I just kicked max out from my commits. 😂

**Comment by maxyinger** [comment]
@googlebot I consent.

**Comment by maxyinger** [comment]
Kicked out or not, I still consent lol

**Pull request merged**